app/assets/javascripts/validation.js

```diff
@@ -1,4 +1,23 @@
 (() => {
+    const messages = (key, interpolation) => {
+        const enMessages = {
+            "uploadFileComponent.error.nameStart": "File name must start with a letter or number, and only contain hyphen, underscore or dot as special characters",
+            "uploadFileComponent.error.fileSize" : `File size must not be bigger than ${interpolation} Megabytes (MB)`,
+            "uploadFileComponent.error.extension" : `File must have an extension of ${interpolation}`,
+            "uploadFileComponent.selectFile" : "Select a file",
+            "uploadFileComponent.error.emptyFile": "The selected file is empty"
+        }
+        const cyMessages = {
+            "uploadFileComponent.error.nameStart": "Mae’n rhaid i enw’r ffeil ddechrau gyda llythyren neu rif, a chynnwys cysylltnod, tanlinell, neu ddot yn unig fel cymeriadau arbennig",
+            "uploadFileComponent.error.fileSize" : `Mae’n rhaid i faint y ffeil beidio â bod yn fwy na ${interpolation} Megabeit (MB)`,
+            "uploadFileComponent.error.extension" : `Mae’n rhaid i estyniad y ffeil fod yn ${interpolation}`,
+            "uploadFileComponent.selectFile" : "Dewiswch ffeil",
+            "uploadFileComponent.error.emptyFile": "Mae'r ffeil a ddewiswyd yn wag"
+        }
+        const lang = document.documentElement.getAttribute("lang")
+        if (lang === "cy") return cyMessages[key]
+        else return enMessages[key]
+    }
```

Comment:
Would be better not to have these translations in the js file (as their existence would be easily missed being separate from the i18n message files. 
Could you not get the java script to look up the message text from some hidden DOM element instead?